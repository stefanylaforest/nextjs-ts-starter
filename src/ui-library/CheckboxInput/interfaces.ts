export default interface CheckboxInputProps {
  name: string;
  label: string | React.ReactNode;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  checked: boolean;
  disabled?: boolean;
}
