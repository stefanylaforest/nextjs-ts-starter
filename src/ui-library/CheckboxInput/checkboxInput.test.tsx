import React, { useState } from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import CheckboxInput from './checkboxInput';

describe('Checkbox Input', () => {
  it('should have checked property', async () => {
    render(
      <CheckboxInput name={'test'} label={'my checkbox label'} onChange={() => {}} checked={true} />
    );
    expect(await screen.findByRole('checkbox', { name: 'my checkbox label' })).toBeChecked();
  });

  it('should set the input to checked when clicked', async () => {
    const Wrapper = () => {
      const [checked, setChecked] = useState<boolean>(false);
      return (
        <CheckboxInput
          name={'test'}
          label={'this is a test'}
          onChange={() => setChecked(!checked)}
          checked={checked}
        />
      );
    };
    render(<Wrapper />);
    const input = await screen.findByRole('checkbox', { name: 'this is a test' });
    await userEvent.click(input);
    expect(input).toBeChecked();
  });

  it('should render without crashing', () => {
    render(
      <CheckboxInput name={'test'} label={'this is a test'} onChange={() => {}} checked={true} />
    );

    expect(screen.findByRole('checkbox', { name: 'this is a test' })).toMatchSnapshot();
  });
});
