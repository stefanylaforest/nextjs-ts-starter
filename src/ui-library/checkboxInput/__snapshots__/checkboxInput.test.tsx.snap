// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Input should render without crashing 1`] = `Promise {}`;
